File: online_judges/codeforces/D/702D.cpp
#include<bits/stdc++.h>

using namespace std;
typedef long long ll;
const ll OO = 1000000000000000000;
ll MIN(ll a ,ll b){
	return (a<=b)?a:b;
}
int main(){
	// freopen("i","r",stdin);
	ll d,a,b,k,t,ans,n;
	scanf("%I64d %I64d %I64d %I64d %I64d",&d,&k,&a,&b,&t);
	n = d/k;
	ans = d*b;
	if(d<=k)
		ans = ans = d*a;
	else if(t+a*k>k*b)
		ans = k*a + (d - k)*b;
	else
		ans = n*k*a + (n-1)*t + MIN(t + (d%k)*a , (d%k)*b);
	printf("%I64d\n",ans);
}

File: online_judges/codeforces/D/702D.py
d,k,a,b,t = [int(s) for s in input().split()] 

ans = d*b
n = d//k

if d<=k:
    ans = ans = d*a
elif t+a*k>k*b :
    ans = k*a + (d - k)*b
else :
    ans = n*k*a + (n-1)*t + min(t + (d%k)*a , (d%k)*b)
print(ans)
